pinot-spi/src/main/java/org/apache/pinot/spi/utils/JsonUtils.java

```diff
@@ -719,4 +719,18 @@ private static void addFieldToPinotSchema(Schema pinotSchema, DataType dataType,
       }
     }
   }
+
+  public static List<Map<String, String>> flatten(String jsonString, JsonIndexConfig jsonIndexConfig) throws IOException {
+    JsonNode jsonNode;
+    try {
+      jsonNode = JsonUtils.stringToJsonNode(jsonString);
+    } catch (IOException e) {
```

Comment:
Why limit to IOException? Can we catch a `Throwable` here?

Reply:
i think the actual exception here we supposed to cathc is malform json exception only. all other issues should still throw --> e..g we should not return incorrect result or simply replace that with the default when there are other issues.

Reply:
Thanks for the explanation @walterddr ,

Reply:
@ankitsultana @walterddr 
I change the source catch JsonProcessingException because the stringToJsonNode internal invoke the DEFAULT_READER.readTree() which throws JsonProcessingException only.

```
JsonUtils.stringToJsonNode(...) throws IOException {
        DEFAULT_READER.readTree(...);
}

class ObjectReader {
      public JsonNode readTree(String json) throws JsonProcessingException, JsonMappingException;
}
```
JsonMappingException is a child of JsonProcessingException.